ibind/support/py_utils.py

```diff
@@ -309,8 +309,8 @@ def params_dict(required: dict = None, optional: dict = None, preprocessors: dic
     for key, value in optional.items():
         if value is not None and value != [None]:
             if preprocessors is not None and key in preprocessors:
-                value = preprocessors[key](value)
-            d[key] = value
+                processed_value = preprocessors[key](value)
+            d[key] = processed_value
 
```

Comment:
if preprocessors is None or key is not in preprocessors, `processed_value` is not defined at the time of calling:

```python
d[key] = processed_value
```

Reply:
Ah good catch!  I'll make a new temp var to represent `value`  - let me know what you think.